otwcode/otwarchive
# AO3-6359 Preventing deleting pseud that corresponds with username

AO3-6359 Fix Possible to delete the pseud that corresponds with your username if the capitalization or diacritics differ

# Pull Request Checklist

* [yes] Have you read ["How to write the perfect pull request"](https://github.blog/2015-01-21-how-to-write-the-perfect-pull-request/)?
* [yes] Have you read the [contributing guidelines](https://github.com/otwcode/otwarchive/blob/master/CONTRIBUTING.md)?
* [yes] Have you added [tests for any changed functionality](https://github.com/otwcode/otwarchive/wiki/Automated-Testing)?
* [yes] Have you added the [Jira](https://otwarchive.atlassian.net) issue number
  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
* [yes] Do you have fewer than 5 pull requests already open? If not, please wait
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-6359

## Purpose

What does this PR do?
Fixes issue AO3-6359 where it was possible to delete the pseud that corresponds with your username if the capitalization or diacritics differ, this was done by changing the psued updating code when the username is changed from doing nothing if the psued already exists to changing the matching psued to match the new username.

## Testing Instructions

How can the Archive's QA team verify that this is working as you intended?
There is replication instructions on the Jira issue, when done post fix the pseud will now match the new username and will not be able to be deleted.
A test was also added into \otwarchive\spec\models\user_spec.rb to test the change.

If you have a Jira account with access, please update or comment on the issue
with any new or missing testing instructions instead.

## References

Are there other relevant issues/pull requests/mailing list discussions?
No.

## Credit

What name and pronouns should we use to credit you in the [Archive of Our Own's Release Notes](https://archiveofourown.org/admin_posts?tag=1)?
Dobe He/Him
If you have a Jira account, please include the same name in the "Full name"
field on your Jira profile, so we can assign you the issues you're working on.

*Please note that if you do not fill in this section, we will use your GitHub account name and
they/them pronouns.*
